File: sdks/python/bolt_prediction_sdk/models/focal_loss_config.py
"""
FocalLossConfig 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class FocalLossConfig(SDKBaseModel):
    """Focal Loss配置"""

    enabled: Optional[bool] = Field(description="是否启用Focal Loss", default=False)
    gamma: Optional[float] = Field(description="聚焦参数gamma，难例加权系数", default=2.0)
    alpha: Optional[Any] = Field(description="类别权重alpha列表", default=None)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/FocalLossConfig.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** Focal Loss配置 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class FocalLossConfig {

    @JsonProperty("enabled")
    private Boolean enabled;

    @JsonProperty("gamma")
    private Double gamma;

    @JsonProperty("alpha")
    private Object alpha;

    public Boolean getEnabled() {
        return enabled;
    }

    public void setEnabled(Boolean enabled) {
        this.enabled = enabled;
    }

    public Double getGamma() {
        return gamma;
    }

    public void setGamma(Double gamma) {
        this.gamma = gamma;
    }

    public Object getAlpha() {
        return alpha;
    }

    public void setAlpha(Object alpha) {
        this.alpha = alpha;
    }

}